homeassistant/components/eufy.py

```diff
@@ -0,0 +1,76 @@
+"""
+Support for Eufy devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/eufy/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_ADDRESS, \
+    CONF_DEVICES, CONF_USERNAME, CONF_PASSWORD, CONF_TYPE, CONF_NAME
+from homeassistant.helpers import discovery
+
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['lakeside==0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'eufy'
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_ADDRESS): cv.string,
+    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Required(CONF_TYPE): cv.string,
+    vol.Optional(CONF_NAME): cv.string
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA},
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
```

Comment:
Use `vol.Inclusive` to make username/password a pair that is either passed in, or not. [Example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/calendar/caldav.py#L37-L38)